docs/api.rst

```diff
@@ -516,6 +516,50 @@ Setup methods
    GridComponent.add_data_from_geodataframe
 
 
+KernelConfigComponent
+=========
```

Comment:
I think `ConfigComponent` is most consistent with the other names. Basically all `ModelComponent` classes are model or kernel components. I think renaming the HydroMT config to HydroMT workflow would be a better way to avoid confusion with the `ConfigComponent`